src/core/qgsabstractdatabaseproviderconnection.h

```diff
@@ -0,0 +1,406 @@
+      TableExists = 1 << 15,        //!< Can check if table exists
+      Spatial = 1 << 16,            //!< The connection supports spatial tables
+    };
+
+    Q_ENUM( Capability )
+    Q_DECLARE_FLAGS( Capabilities, Capability )
+    Q_FLAG( Capabilities )
+
+    /**
+     * Creates a new connection with \a name by reading its configuration from the settings.
+     * If a connection with this name cannot be found, an empty connection will be returned.
+     */
+    QgsAbstractDatabaseProviderConnection( const QString &name );
+
+    /**
+     * Creates a new connection with \a name and initializes the connection from the \a uri.
+     * The connection is not automatically stored in the settings.
+     * \see store()
+     */
+
+    QgsAbstractDatabaseProviderConnection( const QString &name, const QString &uri );
+
+    // Public interface
+
+    /**
+     * Returns connection capabilities
+     */
+    Capabilities capabilities() const;
+
+    // Operations interface
+
+    /**
+     * Creates an empty table with \a name in the given \a schema (schema is ignored  if not supported by the backend)
+     */
+    virtual void createVectorTable( const QString &schema, const QString &name, const QgsFields &fields, QgsWkbTypes::Type wkbType, const QgsCoordinateReferenceSystem &srs, bool overwrite, const QMap<QString, QVariant> *options ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Checks whether a table \a name exists in the given \a schema
+     * \throws QgsProviderConnectionException
+     */
+    virtual bool tableExists( const QString &schema, const QString &name ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Drops a vector (or aspatial) table with given \a schema (schema is ignored if not supported by the backend) and \a name
+     * \note it is responsibility of the caller to handle opened layers and registry entries.
+     * \throws QgsProviderConnectionException
+     */
+    virtual void dropVectorTable( const QString &schema, const QString &name ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Drops a raster table with given \a schema (schema is ignored  if not supported by the backend) and \a name
+     * \note it is responsibility of the caller to handle opened layers and registry entries.
+     * \throws QgsProviderConnectionException
+     */
+    virtual void dropRasterTable( const QString &schema, const QString &name ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Renames a table with given \a schema (schema is ignored  if not supported by the backend) and \a name
+     * \note it is responsibility of the caller to handle opened layers and registry entries.
+     * \throws QgsProviderConnectionException
+     */
+    virtual void renameTable( const QString &schema, const QString &name, const QString &newName ) const SIP_THROW( QgsProviderConnectionException );
+
+    /**
+     * Creates a new schema \a schema
```

Comment:
```suggestion
     * Creates a new schema with the specified \a name
```

What happens if the schema already exists?

Reply:
An exception is raised. I can add a test for it.

Reply:
I was commenting more on adding this note to the dox